Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -243,6 +242,40 @@ class OfflineStoreKit1IntegrationTests: BaseOfflineStoreKitIntegrationTests {
         )
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testCallToGetCustomerInfoWithPendingTransactionsPostsReceiptOnlyOnce() async throws {
+        // This test requires the "production" behavior to make sure
+        // we don't refresh the receipt a second time when posting the second transaction.
+        self.enableReceiptFetchRetry = false
+
+        self.serverDown()
+
+        try await self.purchaseMonthlyProduct()
+        try self.testSession.forceRenewalOfSubscription(
+            productIdentifier: await self.monthlyPackage.storeProduct.productIdentifier
+        )
+
+        try await asyncWait(description: "Expected 2 unfinished transactions") {
+            await Transaction.unfinished.extractValues().count == 2
+        }
+
+        self.serverUp()
+
+        let customerInfo = try await self.purchases.customerInfo(fetchPolicy: .fetchCurrent)
+        try await self.verifyEntitlementWentThrough(customerInfo)
+
+        self.logger.verifyMessageWasLogged(
+            "Found 2 unfinished transactions, will post receipt in lieu of fetching CustomerInfo",
+            level: .debug,
+            expectedCount: 1
+        )
+
+        try await self.logger.verifyMessageIsEventuallyLogged(
```

Comment:
this approach guarantees that the block itself will be de-duped on multiple getCustomerInfo calls, do we have a test to ensure that the first and second one get de-duped? If the message got logged more than once, this test would still pass, right?

Reply:
What am I saying, the message is literally about de-duping. Ignore me.